pkgs/by-name/pr/proton-mail/package.nix

```diff
@@ -0,0 +1,54 @@
+{ lib
+, stdenv
+, fetchurl
+, makeWrapper
+, dpkg
+, electron
+}
+:
+stdenv.mkDerivation rec {
+  pname = "proton-mail";
+  versionBase = "1.0.0";
+  versionSuffix = "beta";
+  version = "${versionBase}-${versionSuffix}";
+
+  src = fetchurl {
+    url = "https://github.com/ProtonMail/inbox-desktop/releases/download/v${versionBase}/${pname}_${versionBase}_amd64.deb";
+    hash = "sha256-3FRm02Ew9GuGMXFAkwSL+FbeLyaI1SmpDU3A5eJh6TU=";
+  };
+
+  dontConfigure = true;
+  dontBuild = true;
+
+  nativeBuildInputs = [ dpkg makeWrapper ];
+
+  unpackPhase = "dpkg-deb --fsys-tarfile $src | tar -x --no-same-permissions --no-same-owner";
+
+  installPhase = ''
+    mkdir -p $out
+    cp -r usr/share/ $out/
+    cp -r usr/lib/proton-mail/resources/app.asar $out/share/
+    rm -rf usr
+    runHook postInstall
+  '';
+
+  preFixup = ''
+    makeWrapper ${electron}/bin/electron $out/bin/${pname} \
+      --add-flags $out/share/app.asar \
+      --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform-hint=auto --enable-features=WaylandWindowDecorations}}" \
+      --set-default ELECTRON_FORCE_IS_PACKAGED 1 \
+      --set-default ELECTRON_IS_DEV 0 \
+      --inherit-argv0
+  '';
+
+  meta = with lib; {
+    description = "Desktop application for Mail and Calendar, made with Electron";
+    homepage = "https://github.com/ProtonMail/inbox-desktop";
+    license = licenses.gpl3Plus;
```

Comment:
Is it currently possible to obtain the source code under this license?

Reply:
The source is not hosted anywhere but GPL3 doesn't require that. The distributor of the SW must make it available to the clients, not the general public. There are open issues on Proton's GH with the question about the source code but no replies yet. Previous versions of the desktop app had sources released AFAIK. The licence is taken from the proton's github.

Reply:
I know. Those issues show that it's not *actually* possible to obtain the source code. Thus I don't believe this license declaration.

Reply:
I agree, I'm also skeptical. But it's not my place to declare other licence than the upstream. So unless proton team gives me a response with different license, this has to stay for now.

Reply:
> So unless proton team gives me a response with different license, this has to stay for now.

Why? We can mark it unfree until it's possible to obtain the source.